hypershift-operator/controllers/hostedcluster/hostedcluster_webhook.go

```diff
@@ -221,20 +252,48 @@ func (v nodePoolValidator) validateCreateKubevirtNodePool(ctx context.Context, n
 		return nil, fmt.Errorf("failed to retrive HostedCluster %s/%s; %w", np.Namespace, np.Spec.ClusterName, err)
 	}
 
-	return kvValidator.validate(ctx, v.client, hc)
+	return kvValidator.validate(ctx, hc, np.Spec.Release.Image)
+}
+
+func (v nodePoolValidator) validateUpdateKubevirtNodePool(ctx context.Context, oldNP, newNP *hyperv1.NodePool) error {
+	err := validateJsonAnnotation(newNP.Annotations)
+	if err != nil {
+		return err
+	}
+
+	if oldNP.Spec.Release.Image != newNP.Spec.Release.Image {
+		hc := &hyperv1.HostedCluster{}
+		err = v.client.Get(ctx, client.ObjectKey{Name: newNP.Spec.ClusterName, Namespace: newNP.Namespace}, hc)
+		if err != nil {
+			return fmt.Errorf("failed to retrive HostedCluster %s/%s; %w", newNP.Namespace, newNP.Spec.ClusterName, err)
+		}
+
+		err = kvValidator.validateReleaseImage(ctx, hc, newNP.Spec.Release.Image)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
 }
 
 type kubevirtClusterValidator struct {
-	clientMap kubevirtexternalinfra.KubevirtInfraClientMap
+	client                client.Client
+	clientMap             kubevirtexternalinfra.KubevirtInfraClientMap
+	imageMetaDataProvider hyperutil.ImageMetadataProvider
 }
 
-func (v kubevirtClusterValidator) validate(ctx context.Context, cli client.Client, hc *hyperv1.HostedCluster) (admission.Warnings, error) {
+func (v *kubevirtClusterValidator) validate(ctx context.Context, hc *hyperv1.HostedCluster, releaseImage string) (admission.Warnings, error) {
+	if v == nil {
+		return nil, errors.New("kubevirt validator is not initialized") // should never happen
+	}
+
```

Comment:
I think this function needs a better name now. This is a validation that only occurs at creation time, and is specifically validating the release version and infra cluster versions.